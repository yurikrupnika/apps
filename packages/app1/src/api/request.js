import request from 'axios';

export default request.create({ baseURL: 'http://localhost:4000' });
