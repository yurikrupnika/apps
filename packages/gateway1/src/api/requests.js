import axios from 'axios';
import { baseURL } from '../config';

const service0 = 4000;
const service1 = 4001;
const service2 = 4002;
const service3 = 4003;

const service0Request = axios.create({ baseURL: `${baseURL}${service0}` });
const service1Request = axios.create({ baseURL: `${baseURL}${service1}` });
const service2Request = axios.create({ baseURL: `${baseURL}${service2}` });
const service3Request = axios.create({ baseURL: `${baseURL}${service3}` });

export {
    service0Request,
    service1Request,
    service2Request,
    service3Request
};
