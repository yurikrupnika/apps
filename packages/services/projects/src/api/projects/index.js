import express from 'express';
import {
    list,
    find,
    removeOne,
    create,
    update,
    schema
} from '@krupnik/express-responses';
import { url } from './config';
import Model from './model';

const route = express.Router();

route.get(`${url}/schema`, schema(Model)); // array
route.get(url, list(Model)); // array
route.get(`${url}/:id`, find(Model)); // object
route.post(url, create(Model));
//
route.put(url, update(Model));

route.delete(`${url}/:id`, removeOne(Model)); // id

export default route;
