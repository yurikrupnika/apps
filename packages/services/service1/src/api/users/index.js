import express from 'express';
import {
    list,
    find,
    removeOne,
    create,
    update,
    schema
} from '@krupnik/express-responses';
import { url } from './config';
import Model from './model';

const route = express.Router();

/**
 * @swagger
 * /api/users/schema:
 *   get:
 *     tags:
 *       - Users
 *     description: Get user schema
 *     summary: Gets api schema
 *     security:
 *       - bearerAuth: []
 *     consumes:
 *       - application/json
 *     produces:
 *       - application/json
 *     responses:
 *       200:
 *         description: Schema of an api
 *         content:
 *            application/json; charset=utf-8:
 *              schema:
 *                type: object
 *                $ref: '#/components/schemas/User'
 *       401:
 *         description: No auth token
 */
route.get(`${url}/schema`, schema(Model));

/**
 * @swagger
 * /api/users:
 *   get:
 *     tags:
 *       - Users
 *     description: Get all users
 *     summary: Finds users
 *     security:
 *       - bearerAuth: []
 *     consumes:
 *       - application/json
 *     produces:
 *       - application/json
 *     responses:
 *       200:
 *         description: Schema of an api
 *         content:
 *            application/json; charset=utf-8:
 *              schema:
 *                type: array
 *                items:
 *                  $ref: '#/components/schemas/User'
 *       401:
 *         description: No auth token
 */

route.get(url, list(Model)); // array
/**
 * @swagger
 * /api/users/{id}:
 *   get:
 *     tags:
 *       - Users
 *     name: get user 1
 *     summary: Finds a user
 *     security:
 *       - bearerAuth: []
 *     consumes:
 *       - application/json
 *     produces:
 *       - application/json
 *     parameters:
 *       - in: path
 *         name: id
 *         schema:
 *           type: string
 *         required:
 *           - id
 *     responses:
 *       200:
 *         description: A single user object
 *         content:
 *            application/json; charset=utf-8:
 *              schema:
 *                type: object
 *                $ref: '#/components/schemas/User'
 *       401:
 *         description: No auth token
 */
route.get(`${url}/:id`, find(Model)); // object
/**
 * @swagger
 * /api/users:
 *   post:
 *     tags:
 *       - Users
 *     name: create user
 *     summary: Create a user
 *     security:
 *       - bearerAuth: []
 *     consumes:
 *       - application/json
 *     produces:
 *       - application/json
 *     parameters:
 *       - in: body
 *         name: user
 *         schema:
 *           type: string
 *         required:
 *           - username
 *     responses:
 *       200:
 *         description: A single user object
 *         schema:
 *           $ref: '#/components/schemas/User'
 *       401:
 *         description: No auth token
 */
route.post(url, create(Model));
/**
 * @swagger
 * /api/users:
 *   put:
 *     tags:
 *       - Users
 *     name: create user
 *     summary: Create a user
 *     security:
 *       - bearerAuth: []
 *     consumes:
 *       - application/json
 *     produces:
 *       - application/json
 *     requestBody:
 *       required: true
 *       description: A JSON object that contains the user name and age.
 *       content:
 *          application/json:
 *              schema:
 *                  $ref: '#/components/schemas/User'
 *     responses:
 *       200:
 *         description: A single user object
 *         content:
 *            application/json; charset=utf-8:
 *              schema:
 *                type: array
 *                items:
 *                  $ref: '#/components/schemas/User'
 *       401:
 *         description: No auth token
 */
route.put(url, update(Model));
/**
 * @swagger
 * /api/users/{id}:
 *   delete:
 *     tags:
 *       - Users
 *     name: create user
 *     summary: Create a user
 *     security:
 *       - bearerAuth: []
 *     consumes:
 *       - application/json
 *     produces:
 *       - application/json
 *     parameters:
 *       - in: path
 *         name: id
 *         schema:
 *           type: string
 *         required:
 *           - id
 *     responses:
 *       200:
 *         description: A single user object
 *         content:
 *            application/json; charset=utf-8:
 *              schema:
 *                type: array
 *                items:
 *                  $ref: '#/components/schemas/User'
 *       401:
 *         description: No auth token
 */
route.delete(`${url}/:id`, removeOne(Model)); // id

export default route;
