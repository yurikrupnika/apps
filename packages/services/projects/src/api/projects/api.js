import request from '../request';
import { url, provider } from './config';

const api = {
    provider,
    fetch: (params) => request.get(`/api${url}`, { params })
        .then((res) => res.data)
        .catch((error) => error)
};

export default api;
