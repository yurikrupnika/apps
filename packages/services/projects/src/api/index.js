import express from 'express';
import projects from './projects';

const route = express.Router();

route.use('/api', projects);

export default route;
