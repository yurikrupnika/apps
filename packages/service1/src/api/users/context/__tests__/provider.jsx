
import React from 'react';
import { Provider } from '../index';
import api from '../../api';


jest.mock('../../api'); // eslint-disable-line no-undef

const {
    describe,
    test,
    expect,
    shallow
} = global;

describe('currency provider', () => {
    const Component = () => (
        <div>
            helo
        </div>
    );

    test('static method', () => {
        Provider.provider();
    });

    test('renders Projects Provider', () => {
        const wrapper = shallow((
            <Provider>
                <Component />
            </Provider>
        ));
        expect(wrapper).toMatchSnapshot();
    });

    test('should test fetch', (done) => {
        const wrapper = shallow((
            <Provider>
                <Component />
            </Provider>
        ));
        const instance = wrapper.instance();
        instance.fetch(); // const result = instance.fetch() is undefined
        expect(api.fetch).toHaveBeenCalledTimes(1);
        return done();
    });

    test('should test catch', (done) => {
        const wrapper = shallow((
            <Provider>
                <Component />
            </Provider>
        ));
        const instance = wrapper.instance();
        instance.fetch();
        expect(api.fetch).toHaveBeenCalledTimes(2);
        return done();
    });
});
