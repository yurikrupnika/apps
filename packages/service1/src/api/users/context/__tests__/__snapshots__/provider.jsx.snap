// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currency provider renders Projects Provider 1`] = `
<ContextProvider
  value={
    Object {
      "data": Array [],
      "fetch": [Function],
      "lastFetch": "",
      "loading": false,
      "selected": null,
    }
  }
>
  <Component />
</ContextProvider>
`;
